src/cmds/qmgr.c

```diff
@@ -270,8 +269,7 @@ attrlist_add(struct attropl  **attrlist, char *attname,
 	} else {
 		ltxt = attval_len;
 		Mstring(paol->value, ltxt+1);
-		strncpy(paol->value, attval, ltxt);
-		paol->value[ltxt] = '\0';
+		pbs_strncpy(paol->value, attval, ltxt + 1);
```

Comment:
The attribute functions are probably super frequent calls, and needs to be super efficient. I think if we could avoid strncpy in these, that would be really good. Besides the destination buffer is allocated based on size of the source string anyway, so why do strncpy and not just strcpy? (in fact strdup takes care of the whole thing, no?)

Reply:
Sure, I can remove this pbs_strncpy. The reason I left it there was because strncpy was already getting used before my change and qmgr command deals with one object at a time so there wouldn't be a measurable performance degradation in this command. Now if this problem was in qstat then yes, I completely agree with you.

Reply:
Ah ok it is qmgr, right! I missed that - it is fine then, no chance of any performance issues